crates/core/tedge_api/src/entity_store.rs

```diff
@@ -691,18 +691,13 @@ impl EntityTree {
     /// Recursively remove an entity, its child devices and services
     ///
     /// Populate the given vector with the topic identifiers of the removed entities
-    fn remove(&mut self, topic_id: &EntityTopicId, removed_entities: &mut Vec<EntityTopicId>) {
-        if let Some(children) = self
-            .entities
-            .get(topic_id)
-            .map(|node| node.children.clone())
-        {
+    fn remove(&mut self, topic_id: &EntityTopicId, removed_entities: &mut Vec<EntityMetadata>) {
```

Comment:
The doc comment has to be updated.

```suggestion
    /// Populate the given vector with the removed entities
    fn remove(&mut self, topic_id: &EntityTopicId, removed_entities: &mut Vec<EntityMetadata>) {
```

Reply:
Resolved by 558238f0